draft-normington-jsonpath-latest.xml

```diff
@@ -202,24 +202,32 @@ root-selector = %x24               ; $ selects document root node
                   <name>Syntax</name>
                   <t>A dot child matcher has a key known as a dot child name or a single asterisk (<tt>*</tt>).
                   </t>
-                  <t>A dot child name corresponds to a key in a JSON object, without the surrounding double quotes (<tt>"</tt>).
+                  <t>A dot child name corresponds to a name in a JSON object. It can be unquoted or surrounded with quotes, either double (<tt>"</tt>) or single (<tt>'</tt>).
                   </t>
                   <sourcecode type="abnf">
 matcher = dot-child             ; see below for alternatives
 dot-child = %x2E dot-child-name / ; .&lt;dot-child-name&gt;
             %x2E %x2A             ; .*
-dot-child-name = TBD
+dot-child-name = union-child / ; "key" or 'key' (see below)
+                 unquoted-key
+unquoted-key = 1*(
+                  %x2D / ; -
+                  DIGIT / ; 0-9
+                  ALPHA / ; A-Z / a-z
+                  %x5F / ; _
+                  %x80-10FFFF ; any non-ASCII Unicode character
```

Comment:
I'm not convinced this should include surrogates such as `D800`.

Reply:
I just follow the philosophy of JSON strings here. [Section 7 of JSON RFC](https://tools.ietf.org/html/rfc8259#section-7) defines non-ASCII range as "continuum" of codepoints within Unicode range, and [section 8.2](https://tools.ietf.org/html/rfc8259#section-8.2) explicitly describes the surrogate case.

Reply:
I really like following JSON string syntax. Since it's an area that will be debated again, I think it would be worth adding some text to this document (like the comment above) to call that out.